core/web/frontend/generic.py

```diff
@@ -77,9 +78,8 @@ def new(self, klass=None):
     def edit(self, id):
         obj = self.klass.objects.get(id=id)
         #ToDo Group admins support
-        if hasattr(obj, "create_by") and current_user.username != obj.created_by or \
```

Comment:
I fixed the condition (namely the typo in "create_by" which would have always evaluated this to FALSE, so everything depended on the "or not admin", which would've been true for all non-admin users